### ISSUE TITLE
resolution time has *quadrupled* between 235d77457 and 6fa61b810

### ISSUE THREAD

**Issue by arielb1** [rust-lang/rust#33061]
Before (on librustc):
```
time: 0.013; rss: 169MB language item collection
time: 0.281; rss: 201MB resolution
```

After (on same librustc):
```
time: 0.017; rss: 226MB language item collection
time: 1.055; rss: 277MB resolution
```
cc @jseyfried

**Comment by arielb1** [comment]
The extra RAM usage seems to be because of the AST not getting freed after the early lint checks. Not sure if this is intentional.

**Comment by jonas-schievink** [comment]
The AST should definitely still get freed, nobody touched [the code doing that](https://github.com/rust-lang/rust/blob/412e6f7e84986c8b9216eafb1d952c576312fe3f/src/librustc_driver/driver.rs#L165-L171), so it's not intentional (unless something about the command line options has changed?)

**Comment by jseyfried** [comment]
The performance regression in `resolve` looks like it was caused by #32328. I'm working on a fix now.
I'm not sure what caused the RAM increase.

**Comment by retep998** [comment]
Resolution used to take about 1 seconds a few months ago in winapi and now it takes 9 seconds.

**Comment by jseyfried** [comment]
I fixed the performance regression in #33064.
